src/catalog/src/durable/transaction.rs

```diff
@@ -1801,6 +1802,23 @@ impl<'a> Transaction<'a> {
                 .map(move |(update, diff)| (kind_fn(update), diff))
         }
 
+        fn get_large_collection_op_updates<'a, K, T>(
+            collection: &'a Vec<(K, Diff, Timestamp)>,
+            kind_fn: impl Fn(T) -> StateUpdateKind + 'a,
+            op: Timestamp,
+        ) -> impl Iterator<Item = (StateUpdateKind, Diff)> + 'a
+        where
+            K: Ord + Eq + Clone + Debug,
+            T: DurableType<K, ()>,
+        {
+            collection
+                .iter()
+                .filter(move |(_k, _diff, ts)| *ts == op)
```

Comment:
Is `filter_map` a better choice over `filter` and 3 `map`s?

Reply:
Updated